Integrations/src/test/java/io/deephaven/integrations/learn/GathererTest.java

```diff
@@ -0,0 +1,167 @@
+    };
+    private static final String[] intColNames = {"int1", "int2"};
+    private static final int[][] intData = {
+            new int[] {100, 200, -100, -200},
+            new int[] {-300, -400, 300, 400},
+    };
+    private static final String[] longColNames = {"long1", "long2"};
+    private static final long[][] longData = {
+            new long[] {1L, 100L, 10000L, 1000000L},
+            new long[] {9L, 999L, 99999L, 9999999L},
+    };
+    private static final String[] floatColNames = {"float1", "float2"};
+    private static final float[][] floatData = {
+            new float[] {3.14F, 2.73F, 1.5F, 0.63F},
+            new float[] {0.1F, 0.2F, 0.3F, 0.4F},
+    };
+    private static final String[] doubleColNames = {"double1", "double2"};
+    private static final double[][] doubleData = {
+            new double[] {3.14, 2.73, 1.5, 0.63},
+            new double[] {0.1, 0.2, 0.3, 0.4}
+    };
+    private static final String[] columnNames = new String[] {
+            boolColNames[0], boolColNames[1],
+            byteColNames[0], byteColNames[1],
+            shortColNames[0], shortColNames[1],
+            intColNames[0], intColNames[1],
+            longColNames[0], longColNames[1],
+            floatColNames[0], floatColNames[1],
+            doubleColNames[0], doubleColNames[1]
+    };
+    private static final Object[] columnData = new Object[] {
+            boolData[0], boolData[1],
+            byteData[0], byteData[1],
+            shortData[0], shortData[1],
+            intData[0], intData[1],
+            longData[0], longData[1],
+            floatData[0], floatData[1],
+            doubleData[0], doubleData[1]
+    };
+
+    @BeforeClass
+    public static void setup() {
+        table = new InMemoryTable(columnNames, columnData);
+    }
+
+    public static ColumnSource<?>[] getColSet(final String[] colNames) {
+        ColumnSource<?>[] rst = new ColumnSource[2];
+
+        for (int i = 0; i < 2; i++) {
+            rst[i] = table.getColumnSource(colNames[i]);
+        }
+
+        return rst;
+    }
+
+    private static <T> void assertRowMajor(BiFunction<Integer, Integer, T> expected, Function<Integer, T> actual) {
+        // Data should be stored in row-major order
+        int idx = 0;
+        for (int j = 0; j < 4; j++) {
+            for (int i = 0; i < 2; i++) {
+                Assert.assertEquals("i=" + i + " j=" + j, expected.apply(i, j), actual.apply(idx));
+                idx++;
+            }
+        }
+    }
+
+
+    private static <T> void assertColumnMajor(BiFunction<Integer, Integer, T> expected, Function<Integer, T> actual) {
+        // Data should be stored in column-major order
+        int idx = 0;
+        for (int i = 0; i < 2; i++) {
+            for (int j = 0; j < 4; j++) {
+                Assert.assertEquals("i=" + i + " j=" + j, expected.apply(i, j), actual.apply(idx));
+                idx++;
+            }
+        }
+    }
+
```

Comment:
We either need to remove/comment-out the boolean method or test it here.  We don't want a user to use it if we think it is broken or untested.